code/modules/shuttle/navigation_computer.dm

```diff
@@ -31,21 +33,23 @@
 		actions += place_action
 
 /obj/machinery/computer/camera_advanced/shuttle_docker/CreateEye()
-	var/obj/docking_port/mobile/M = SSshuttle.getShuttle(shuttleId)
-	if(QDELETED(M))
+	shuttle_port = SSshuttle.getShuttle(shuttleId)
+	if(QDELETED(shuttle_port))
+		shuttle_port = null
 		return
+
+	for(var/V in shuttle_port.shuttle_areas)
+		var/area/A = V
+		if(!QDELETED(A))
+			shuttle_port_areas[A] = 1
```

Comment:
Making this extra list costs more than just letting it check in the main loop

Reply:
Making this list is done once, when the eye is created. The check is done on every turf to be checked every time the eye moves, which is considerably more expensive.

Reply:
They *both* only get checked in CreateEye

Reply:
https://github.com/tgstation/tgstation/pull/31722/files#diff-23c148e6e1bfb5037db9d05f3a02c9fdR129
This is the check that I am referring to.